pkg/kubelet/dockershim/remote/docker_service.go

```diff
@@ -0,0 +1,168 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package remote
+
+import (
+	"fmt"
+
+	"golang.org/x/net/context"
+
+	internalApi "k8s.io/kubernetes/pkg/kubelet/api"
+	runtimeApi "k8s.io/kubernetes/pkg/kubelet/api/v1alpha1/runtime"
+	"k8s.io/kubernetes/pkg/kubelet/dockershim"
+)
+
+// DockerService uses dockershim service to implement runtime and image services.
+// Notice that the contexts in the functions are not used now.
+// TODO(random-liu): Change the dockershim service to support context, and implement
+// internal services and remote services with the dockershim service.
+type DockerService struct {
+	runtimeService internalApi.RuntimeService
+	imageService   internalApi.ImageManagerService
+}
```

Comment:
On the other hand, you can make `DockerService` an interface and rename the struct to be `dockerService`. Then you can return the interface in your constructor
 `NewDockerService(s dockershim.DockerService) DockerService`